src/lib/expansion/expansion-panel-header.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  Component,
+  Host,
+  Input,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  trigger,
+  state,
+  style,
+  transition,
+  animate,
+} from '@angular/animations';
+import {MdExpansionPanel, EXPANSION_PANEL_ANIMATION_TIMING} from './expansion-panel';
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-expansion-panel-header, mat-expansion-panel-header',
+  styleUrls: ['./expansion-panel-header.css'],
+  templateUrl: './expansion-panel-header.html',
+  encapsulation: ViewEncapsulation.None,
+  host: {
+    'class': 'mat-expansion-panel-header',
+    'role': 'button',
+    'tabindex': '0',
+    '[attr.aria-controls]': 'getPanelId()',
+    '[attr.aria-expanded]': 'isExpanded()',
+    '[class.mat-expanded]': 'isExpanded()',
+    '(click)': 'changeExpandedState()',
+    '(keyup.space)': 'changeExpandedState()',
+    '(keyup.enter)': 'changeExpandedState()',
```

Comment:
The `keyup.space` syntax unfortunately doesn't work in Safari 9, so we have to manually check the key code in the handler.